eng/common/scripts/Prepare-Release.ps1

```diff
@@ -0,0 +1,295 @@
+
+Write-Host
+Write-Host "Assuming release is in $month with release date $releaseDateString" -ForegroundColor Green
+
+$isNew = "True";
+$libraryType = "Beta";
+$latestVersion = $null;
+foreach ($existingVersion in $existing)
+{
+    $isNew = "False"
+    $parsedVersion = [AzureEngSemanticVersion]::new($existingVersion)
+    if (!$parsedVersion.IsPrerelease)
+    {
+        $libraryType = "GA"
+    }
+
+    $latestVersion = $existingVersion;
+}
+
+$currentProjectVersion = $packageProperties.Version
+
+if ($latestVersion)
+{
+    Write-Host
+    Write-Host "Latest released version $latestVersion, library type $libraryType" -ForegroundColor Green
+}
+else
+{
+    Write-Host
+    Write-Host "No released version, library type $libraryType" -ForegroundColor Green
+}
+
+$newVersion = Read-Host -Prompt "Input the new version, NA if you are not releasing, or press Enter to use use current project version '$currentProjectVersion'"
+$releasing = $true
+
+if (!$newVersion)
+{
+    $newVersion = $currentProjectVersion;
+}
+
+if ($newVersion -eq "NA")
+{
+    $releasing = $false
+}
+
+if ($releasing)
+{
+    if ($latestVersion)
+    {
+        $releaseType = "None";
+        $parsedNewVersion = [AzureEngSemanticVersion]::new($newVersion)
+        if ($parsedNewVersion.Major -ne $parsedVersion.Major)
+        {
+            $releaseType = "Major"
+        }
+        elseif ($parsedNewVersion.Minor -ne $parsedVersion.Minor)
+        {
+            $releaseType = "Minor"
+        }
+        elseif ($parsedNewVersion.Patch -ne $parsedVersion.Patch)
+        {
+            $releaseType = "Patch"
+        }
+        elseif ($parsedNewVersion.IsPrerelease)
+        {
+            $releaseType = "Patch"
+        }
+    }
+    else
+    {
+        $releaseType = "Major";
+    }
+
+    Write-Host
+    Write-Host "Detected released type $releaseType" -ForegroundColor Green
+
+    Write-Host
+    Write-Host "Updating versions to $newVersion with date $releaseDateString" -ForegroundColor Green
+
+    SetPackageVersion -PackageName $package -Version $newVersion -ServiceName $serviceDirectory -ReleaseDate $releaseDateString `
+    -BuildType $BuildType $GroupName $packageProperties.Group
+
+    $fields = @{
+        "Permalink usetag"="https://github.com/Azure/azure-sdk-for-net/sdk/$serviceDirectory/$package/README.md"
+        "Package Registry Permalink"="https://nuget.org/packages/$package/$newVersion"
+        "Library Type"=$libraryType
+        "Release Type"=$releaseType
+        "Version Number"=$newVersion
+        "Planned Release Date"=$releaseDateString
+        "New Library Only"=$isNew
+    }
+    $state = "Active"
+}
+else
+{
+    $fields = @{}
+    $state = "Not In Release"
+}
+
+$workItems = az boards query @commonParameter --project Release --wiql "SELECT [ID], [State], [Iteration Path], [Title] FROM WorkItems WHERE [State] <> 'Closed' AND [Iteration Path] under 'Release\2020\$month' AND [Title] contains '.NET'" | ConvertFrom-Json;
```

Comment:
It is probably worth deleting or commenting out all the work-item creation parts of this script. It will make your testing a lot easier.